File: ransomNote.py
  def canConstruct(self, ransomNote, magazine):
    arr = [0] * 26
    for c in magazine:
        arr[ord(c) - ord('a')] += 1
    for c in ransomNote:
        if arr[ord(c) - ord('a')] == 0:
            return False
        arr[ord(c) - ord('a')] -= 1
    return True

File: ransomNote.js
const canConstruct = (ransomNote, magazine) => {
    let arr = Array(26).fill(0);
    for (const c of magazine) {
        arr[c.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }
    for (const c of ransomNote) {
        if (--arr[c.charCodeAt(0) - 'a'.charCodeAt(0)] < 0) {
            return false;
        }
    }
    return true;
};
